File: python/vetoresCaracteres.py
# -*- coding: utf-8 -*-

# Programa Python demonstra como declarar e navegar vetores de caracteres.
# Um dos diferenciais do Python é a flexibilidade em lidar com vetores.
#
# Objetivos:
#   - uso de arrays que armazenam caracteres;
#   - uso do comando for para preencher arrays;
#   - uso da função xrange para definir os limites do for;
#
# Módulo: Lógica de programação
# Author: Eder Magalhães

vogais = [ 'a', 'e', 'i', chr(111), 'u' ];

# imprime o array de forma natural
print vogais;

# array inicialmente está vazio, é carregado no for abaixo
alfabetoMaiusculo = [];

for l in xrange(ord('Z'), ord('A')-1, -1):
    alfabetoMaiusculo.append(chr(l));

# usa a função join da string pra concatenar o conteúdo do array
print "Conteúdo do array alfabetoMaiusculo: ", ' '.join(alfabetoMaiusculo);

palavras = ["Programando", "arrays", "em", "Python"];

print "Conteúdo do array palavras: ", ' '.join(palavras);

File: java/VetoresCaracteres.java
/**
 * Programa Java demonstra como declarar e navegar vetores de caracteres.
 *
 * Objetivos:
 *   - uso de arrays que armazenam caracteres (char e String);
 *   - uso do comando for para preencher e percorrer arrays;
 *   - todo array possui a propriedade length, que indica q qtde de elementos armazenados;
 *
 * Módulo: Lógica de programação | Capítulo: XXX | Página: XXX
 * Author: Eder Magalhães
 */
public class VetoresCaracteres {

    public static void main(String[] args) {
        char[] vogais = { 'a', 'e', 'i', 111, 'u' }; //111 é 'o' em ASCII

        //com println é possível imprimir os elementos de um array char
        System.out.println(vogais);

        /*
         * Trecho a seguir cria um vetor que armazena o alfabeto, 
         * com as letras Maísculas em ordem decrescente (de Z a A).
         */
        char[] alfabetoMaiusculo = new char[27];
        char letra = 'Z';
        int i = 0;

        while(letra >= 'A') {
           alfabetoMaiusculo[i++] = letra--;
        }

        //trecho para imprimir o conteudo do array alfabetoMaiusculo
        System.out.print("Conteúdo do array alfabetoMaiusculo: ");
        for (char l: alfabetoMaiusculo) {
            System.out.print(l+" ");
        }
        System.out.println();

        String[] palavras = new String[] {"Programando", "arrays", "em", "Java"};
        System.out.print("Conteúdo do array palavras: ");

        //foreach, mais simples
        for (String s: palavras) {
            System.out.print(s+" ");
        }
        System.out.println();
    }

}
